streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -92,32 +89,62 @@ private JoinWindows(final long beforeMs,
         if (beforeMs + afterMs < 0) {
             throw new IllegalArgumentException("Window interval (ie, beforeMs+afterMs) must not be negative.");
         }
+
+        if (graceMs < 0) {
+            throw new IllegalArgumentException("Grace period must not be negative.");
+        }
+
         this.afterMs = afterMs;
         this.beforeMs = beforeMs;
         this.graceMs = graceMs;
         this.enableSpuriousResultFix = enableSpuriousResultFix;
     }
 
+    /**
+     * Specifies that records of the same key are joinable if their timestamps are within {@code timeDifference},
+     * i.e., the timestamp of a record from the secondary stream is max {@code timeDifference} earlier or later than
```

Comment:
Thanks @mjsax for the feedback. I will open a JIRA item to address the fix for the Javadocs.

@ableegoldman @cadonna @showuon please can we approve it today so that we don't miss the cutoff?

I will work on completing any javadocs and unit tests corrections/edits/additions by the end of next week.